### PR TITLE
Support upvote downvote of store entries

### PR DESCRIPTION
**Pull request by Siddharth11**
voting

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -52,3 +52,8 @@ export const SAFETY_CHECKS_LIST = [
   "adequate_social_distancing",
   "home_delivery",
 ];
+
+export const VOTE = {
+  UP: "UP",
```
Comment (arjun27) [inline src/constants.js:57]: i was using lowercase in the backend. can we change to that?

File: src/Homepage/UpvoteDownvote.js [hunk 2]
```diff
@@ -24,12 +25,16 @@ class UpvoteDownvote extends Component {
...
+      },
     };
 
     this.getVoteKey = this.getVoteKey.bind(this);
     this.handleVote = this.handleVote.bind(this);
+    this.handleUnselection = this.handleUnselection.bind(this);
+    this.handleNewSelection = this.handleNewSelection.bind(this);
   }
 
   getVoteKey(entry) {
```
Comment (arjun27) [inline src/Homepage/UpvoteDownvote.js:40]: should this use `update id` and not store id?

File: src/Homepage/UpvoteDownvote.js [hunk 3]
```diff
@@ -45,19 +50,73 @@ class UpvoteDownvote extends Component {
...
+          ...prevState.voteCount,
+          [voteType]: prevState.voteCount[voteType] - 1,
+        },
+        voteValue: null,
+      };
+    });
+
+    this.props.removeItemFromStorage(this.state.voteKey);
+
+    deleteVote(payload);
```
Comment (arjun27) [inline src/Homepage/UpvoteDownvote.js:77]: hmm ideally these 3 actions should be atomic, but we can live with this

File: src/Homepage/UpvoteDownvote.js (hunk 2 above)
Comment (Siddharth11) [reply src/Homepage/UpvoteDownvote.js:40]: yes

**Pull request merged**